reference/filter/filters.xml

```diff
@@ -0,0 +1,774 @@
+        </entry>
+        <entry>
+         <para>
+          当值为 "1"、 "true"、 "on" 和 "yes" 时返回 &true;
+          其他情况返回 &false;
+         </para>
+         <para>
+          如果设置了 <constant>FILTER_NULL_ON_FAILURE</constant>，只有当值为 "0"、
+          "false"、"off"、"no"、"" 返回 &false;。所有非布尔值都返回 &null;
+         </para>
+        </entry>
+       </row>
+       <row>
+        <entry><constant>FILTER_VALIDATE_DOMAIN</constant></entry>
+        <entry>"validate_domain"</entry>
+        <entry>
+         <parameter>default</parameter>
+        </entry>
+        <entry>
+         <constant>FILTER_FLAG_HOSTNAME</constant>
+        </entry>
+        <entry>
+         <para>
+          验证域名长度是否有效
+         </para>
+         <para>
+          根据 RFC 1034, RFC 1035, RFC 952, RFC 1123, RFC 2732, RFC 2181 和 RFC 1123 验证域名。
+          可选标志 <constant>FILTER_FLAG_HOSTNAME</constant> 增加了验证主机名的能力（主机名必须以字母数字字符开头，并且只
+          包含字母数字或连字符）。
+         </para>
+        </entry>
+       </row>
+       <row>
+        <entry><constant>FILTER_VALIDATE_EMAIL</constant></entry>
+        <entry>"validate_email"</entry>
+        <entry>
+         <parameter>default</parameter>
+        </entry>
+        <entry>
+         <constant>FILTER_FLAG_EMAIL_UNICODE</constant>
+        </entry>
+        <entry>
+         <para>
+          验证是否是有效的邮箱地址
+         </para>
+         <para>
+          通常，这将根据 RFC 822 中的语法验证电子邮件地址，但不支持注释、空白和没有点号的域名。
+         </para>
+        </entry>
+       </row>
+       <row>
+        <entry><constant>FILTER_VALIDATE_FLOAT</constant></entry>
+        <entry>"float"</entry>
+        <entry>
+         <parameter>default</parameter>,
+         <parameter>decimal</parameter>,
+         <parameter>min_range</parameter>,
+         <parameter>max_range</parameter>
+        </entry>
+        <entry>
+         <constant>FILTER_FLAG_ALLOW_THOUSAND</constant>
+        </entry>
+        <entry>验证值是否为浮点数，可以从指定范围中选择，成功后转换为浮点类型。</entry>
+       </row>
+       <row>
+        <entry><constant>FILTER_VALIDATE_INT</constant></entry>
+        <entry>"int"</entry>
+        <entry>
+         <parameter>default</parameter>,
+         <parameter>min_range</parameter>,
+         <parameter>max_range</parameter>
+        </entry>
+        <entry>
+         <constant>FILTER_FLAG_ALLOW_OCTAL</constant>,
+         <constant>FILTER_FLAG_ALLOW_HEX</constant>
+        </entry>
+        <entry>验证值是否为整数，可以从指定范围中选择，成功后转换为整数类型。</entry>
+       </row>
+       <row>
+        <entry><constant>FILTER_VALIDATE_IP</constant></entry>
+        <entry>"validate_ip"</entry>
+        <entry>
+         <parameter>default</parameter>
+        </entry>
+        <entry>
+         <constant>FILTER_FLAG_IPV4</constant>,
+         <constant>FILTER_FLAG_IPV6</constant>,
+         <constant>FILTER_FLAG_NO_PRIV_RANGE</constant>,
+         <constant>FILTER_FLAG_NO_RES_RANGE</constant>
+        </entry>
+        <entry>
+         验证值是否是 IP 地址，可选 IPV4、IPV6 或者不从私有地址或保留地址中选择。
```

Comment:
Validates value as IP address, optionally only IPv4 or IPv6 or not from private or reserved ranges.